src/NServiceBus.Core.Analyzer.Tests/AwaitOrCaptureTasksAnalyzerTests.cs

```diff
@@ -0,0 +1,84 @@
+namespace NServiceBus.Core.Analyzer.Tests
+{
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Helpers;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class AwaitOrCaptureTasksAnalyzerTests : DiagnosticVerifier
```

Comment:
Would be great to see message session tests as well. Maybe also on non task returning API and in deeper hierarchies?